dashboard/app/views/levels/editors/_all.html.haml

```diff
@@ -0,0 +1,210 @@
+          %td
+            %input{class: 'y', type: 'number', style: 'width: 80px;'}
+          %td
+            %input{class: 'classes', type: 'text', placeholder: 'no-tip', style: 'width: 60px;'}
+      %button{class: 'remove_callout', type: 'button'} Remove callout
+      %hr
+  %div{id: 'all_callouts_editor'}
+  %button{id: 'add_callout', type: 'button'} Add a callout
+.field
+  = f.label :callout_json
+  %div{style: '-webkit-user-select: text'}
+  = f.text_area :callout_json, rows: 4
+:javascript
+  // <![CDATA[
+    var callout_editor = CodeMirror.fromTextArea($('#level_callout_json').get(0), {
+      mode: 'javascript',
+      viewportMargin: Infinity,
+      matchBrackets: true
+    });
+
+    var calloutJSON = callout_editor.getValue();
+    var calloutModel = {
+      localization_key: "",
+      callout_text: "",
+      element_id: "",
+      on: "",
+      qtip_config: {
+        style: {classes: ""},
+        position: {
+          my: "",
+          at: "",
+          adjust: {x: 0, y: 0}
+        }
+      }
+    };
+
+    // Create callout spaces for each callout in the original JSON
+    if (calloutJSON.length > 0) {
+      var callouts_to_update = JSON.parse(callout_editor.getValue());
+      $.each(callouts_to_update, function (index, callout) {
+        updateTemplate(callout, $createNewCalloutSpace());
+      });
+    }
+
+    /**
+     * For each key in the given model, set the <input> with a matching class name to the key's value.
+     * @param {model} The model to use when updating the DOM.
+     * @param {$template} The jQuery element to search for <input> elements.
+     */
+    function updateTemplate(model, $template) {
+      $.each(model, function (key, value) {
+        if (value && typeof value === 'object') {
+          updateTemplate(value, $template);
+        } else {
+          $template.find('.' + key).val(model[key]);
+        }
+      });
+    }
+
+    /**
+     * For each key in the given model, set the key's value to the value of the <input> with a matching class name.
+     * @param {model} The model to update from the DOM.
+     * @param {$template} The jQuery element to search for <input> elements.
+     */
+    function updateModel(model, $template) {
+      $.each(model, function (key, value) {
+        if (typeof value === 'object') {
+          updateModel(value, $template);
+        } else {
+          var value = $template.find('.' + key).val();
+          model[key] = typeof model[key] === 'number' ? +value : value;
+        }
+      });
+    }
+
+    function updateCalloutJSON() {
+      var updated_callouts = [];
+      $('#all_callouts_editor .callout_space').each(function () {
+        var model = $.extend(true, {}, calloutModel);
+        updateModel(model, $(this));
+        updated_callouts.push(model);
+      });
+      callout_editor.setValue(JSON.stringify(updated_callouts, null, ' '));
+    }
+
+    function $createNewCalloutSpace() {
```

Comment:
Just to show it returns a jQuery object (I've been using the same convention for naming variables).  I can remove it if you prefer.